compiler/rustc_hir/src/def.rs

```diff
@@ -266,9 +266,13 @@ impl DefKind {
             | DefKind::TyAlias
             | DefKind::ForeignTy
             | DefKind::TraitAlias
-            | DefKind::AssocTy
             | DefKind::TyParam
-            | DefKind::ExternCrate => DefPathData::TypeNs(name.unwrap()),
+            | DefKind::ExternCrate => DefPathData::TypeNs(Some(name.unwrap())),
```

Comment:
why not just `name` here? I guess you want to make the thing explode if is `None`?

Reply:
Yes. At the moment a `None` name only makes sense for `DefKind::AssocTy`, so I don't want to accept other combinations that aren't necessary.